OpenImis.ePayment/Escape/Payment/Data/GepgUtility.cs

```diff
@@ -206,245 +187,157 @@ public String CreateGePGCancelPaymentRequest(IConfiguration Configuration, int P
+                    {
+                        gepgBillCanclReq = gepgBillCanclReq,
+                        gepgSignature = signature
+                    };
 
-                xs.Serialize(xw, GePGPaymentCancelRequest, ns);
-                xw.Flush();
-                outString = sb.ToString();
+                    xs = new XmlSerializer(typeof(GePGPaymentCancelRequest));
+                    settings = new XmlWriterSettings();
+                    sb = new StringBuilder();
 
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
-            {
-                if (xw != null)
+                    using (var xw2 = XmlWriter.Create(sb, settings))
+                    {
+                        xs.Serialize(xw, gepgBillCanclReq, ns);
+                        xw.Flush();
+                    }
+                }
+                catch (Exception ex)
                 {
-                    xw.Close();
+                    Console.WriteLine(ex.ToString());
                 }
+
+                return sb.ToString();
             }
-            return outString;
         }
 
         public string FinaliseSignedMsg(string sign)
         {
-
             object gepgData = null;
             gepgData = new GepgBillMessage() { gepgBillSubReq = newBill, gepgSignature = sign };
 
+            var xs = new XmlSerializer(typeof(GepgBillMessage));
+            var ns = new XmlSerializerNamespaces();
+            ns.Add("", "");
+            var settings = new XmlWriterSettings();
+            var sb = new StringBuilder();
 
-            XmlSerializer xs = null;
-            XmlSerializerNamespaces ns = null;
-            XmlWriterSettings settings = null;
-            XmlWriter xw = null;
-            String outString = String.Empty;
-
-            try
-            {
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-                settings = new XmlWriterSettings();
-                //settings.Indent = true;
-                StringBuilder sb = new StringBuilder();
-                xs = new XmlSerializer(typeof(GepgBillMessage));
-
-                xw = XmlWriter.Create(sb, settings);
-
-                xs.Serialize(xw, gepgData, ns);
-                xw.Flush();
-                outString = sb.ToString();
-
-                
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-                if (xw != null)
+                try
                 {
-                    xw.Close();
+                    xs.Serialize(xw, gepgData, ns);
+                    xw.Flush();
                 }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.ToString());
```

Comment:
Instead of logging the error in Console, can we log it to Log4Net or somewhere else to review the exception later?

Reply:
The same as above, i didn't want to modify the component to allow logging